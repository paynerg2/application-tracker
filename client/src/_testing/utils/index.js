import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../../_reducers';
import { middlewares } from '../../app/store';

export const testStore = (initialState) => {
    const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
    return createStoreWithMiddleware(rootReducer, initialState);
};
