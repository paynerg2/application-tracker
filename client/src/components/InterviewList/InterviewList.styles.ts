import styled from 'styled-components';
import Button from '../../components/Button/button';
import { List } from '../List/list';

export const CircularButton = styled(Button)`
    border-radius: 50%;
    height: 1.25em;
    width: 1.25em;
    font-size: 4em;

    &:hover {
        transform: rotate(180deg) scale(1.02);
        transition: all 0.3s ease-in-out;
    }

    @media (max-width: ${(props) => props.theme.breakpoint.laptop}) {
        display: none;
    }
`;

export const ListContainer = styled(List)`
    height: auto;
    gap: 2vmin;

    > ${CircularButton} {
        align-self: flex-end;
        margin-right: 2.5%;
        margin-top: 2.5%;
    }

    @media (max-width: ${(props) => props.theme.breakpoint.laptop}) {
        border-radius: 0;
    }
`;

export const Response = styled.input.attrs({ type: 'radio' })`
    display: none;

    &:not(:disabled) ~ label {
        cursor: pointer;
    }

    &:disabled ~ label {
        color: ${(props) => props.theme.color.veryLightGray};
        box-shadow: none;
        cursor: not-allowed;
    }

    &:checked + label {
        font-weight: 700;
        color: ${(props) => props.theme.color.contrastText};
        box-shadow: ${(props) => props.theme.borders.shadow};
        background: ${(props) =>
            (props.id === 'rejected' && props.theme.color.error) ||
            (props.id === 'offer' && props.theme.color.mintGreen)};
    }

    @media (max-width: ${(props) => props.theme.breakpoint.laptop}) {
        > ${Button} {
        }
    }
`;

export const Label = styled.label`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    font-size: 1em;
    height: auto;
    background: ${(props) => props.theme.color.background};
    border-radius: ${(props) => props.theme.borders.radius};
    padding: 0.2em;
    box-sizing: border-box;
    border: solid 1px ${(props) => props.theme.color.mainText};
    position: relative;

    @media (max-width: ${(props) => props.theme.breakpoint.laptop}) {
    }
`;

export const ResponseGroup = styled.div`
    width: min-content;
    height: auto;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-end;
    gap: 5em;

    @media (max-width: ${(props) => props.theme.breakpoint.laptop}) {
    }
`;

export const Placeholder = styled.div`
    width: 60%;
    font-size: 1.25em;
    font-family: ${(props) => props.theme.font.secondary};
`;
