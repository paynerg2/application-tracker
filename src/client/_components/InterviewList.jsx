import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

const InterviewContainer = styled.select`
    width: 88%;
`;

class InterviewList extends Component {
    // TODO: add function to format date/time

    renderList = () => {
        const { loading, interviews } = this.props;
        return (
            <InterviewContainer>
                {!loading &&
                    interviews.map(interview => {
                        const { _id, startTime, location } = interview;
                        return (
                            <option key={_id}>
                                {`${startTime}, ${location}`}
                            </option>
                        );
                    })}
            </InterviewContainer>
        );
    };

    render() {
        return <React.Fragment>{this.renderList()}</React.Fragment>;
    }
}

function mapStateToProps(state) {
    const { interviewList, loading } = state.interviews;
    return {
        loading,
        interviewList
    };
}

const connectedInterviewList = connect(mapStateToProps)(InterviewList);
export { connectedInterviewList as InterviewList };
