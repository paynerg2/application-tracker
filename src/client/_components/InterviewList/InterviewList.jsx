import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';

const InterviewContainer = styled.select`
    width: 88%;
`;
InterviewContainer.displayName = 'InterviewContainer';

class InterviewList extends Component {
    // TODO: add function to format date/time

    renderList = () => {
        const { loading, interviews } = this.props;
        return (
            <InterviewContainer>
                {!loading &&
                    interviews.map(interview => {
                        const { _id, startTime, location } = interview;
                        return (
                            <option key={_id}>
                                {`${startTime}, ${location}`}
                            </option>
                        );
                    })}
            </InterviewContainer>
        );
    };

    render() {
        return <React.Fragment>{this.renderList()}</React.Fragment>;
    }
}

function mapStateToProps(state) {
    const { loading } = state.interviews;
    return {
        loading
    };
}

InterviewList.propTypes = {
    interviews: PropTypes.arrayOf(PropTypes.object),
    loading: PropTypes.bool
};

const connectedInterviewList = connect(mapStateToProps)(InterviewList);
export { connectedInterviewList as InterviewList };
