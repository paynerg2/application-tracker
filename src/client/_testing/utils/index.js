import checkPropTypes from 'check-prop-types';
import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../../_reducers';

export const checkProps = (component, expectedProps) => {
    const propsError = checkPropTypes(
        component.propTypes,
        expectedProps,
        'props',
        component.name
    );
    return propsError;
};
