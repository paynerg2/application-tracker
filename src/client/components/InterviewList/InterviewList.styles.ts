import styled from 'styled-components';
import Button from '../../components/Button/button';

export const CircularButton = styled(Button)`
    border-radius: 50%;
    height: 8vmin;
    width: 8vmin;
    font-size: 4em;
`;
