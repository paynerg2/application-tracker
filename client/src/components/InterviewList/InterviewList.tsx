import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Interview } from '../../interfaces/interviews';
import InterviewCard from '../Cards/Interview/interviewCard';
import { CircularButton, ListContainer, Placeholder } from './InterviewList.styles';

interface Props {
    interviews: Interview[];
    withAdd?: boolean;
}

function InterviewList({ interviews, withAdd = true }: Props) {
    const navigate = useNavigate();

    return (
        <ListContainer>
            {interviews && interviews.length > 0 ? (
                interviews.map((interview) => (
                    <InterviewCard key={interview.id} interview={interview} />
                ))
            ) : (
                <Placeholder>
                    <div>No interviews yet!</div>
                </Placeholder>
            )}
            {withAdd && (
                <CircularButton onClick={() => navigate('/interviews/new')}>+</CircularButton>
            )}
        </ListContainer>
    );
}

export default InterviewList;
