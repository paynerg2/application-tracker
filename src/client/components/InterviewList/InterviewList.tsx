import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Interview } from '../../interfaces/interviews';
import InterviewCard from '../Cards/Interview/interviewCard';
import { CircularButton, ListContainer } from './InterviewList.styles';

interface Props {
    interviews: Interview[];
}

function InterviewList({ interviews }: Props) {
    const navigate = useNavigate();

    return (
        <ListContainer>
            {interviews && interviews.length > 0 ? (
                interviews.map((interview) => (
                    <InterviewCard key={interview.id} interview={interview} />
                ))
            ) : (
                <div>No interviews... yet!</div>
            )}
            <CircularButton onClick={() => navigate('/interviews/new/1')}>+</CircularButton>
        </ListContainer>
    );
}

export default InterviewList;
