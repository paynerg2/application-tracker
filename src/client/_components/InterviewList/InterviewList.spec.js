import React from 'react';
import { shallow } from 'enzyme';

import { InterviewList } from './InterviewList';
import { testStore } from '../../_testing/utils';

const setUp = (props = {}, initialState = {}) => {
    const store = testStore(initialState);
    const component = shallow(
        <InterviewList store={store} {...props} />
    ).dive();
    return component;
};

const props = {
    interviews: [
        {
            startTime: '5:55PM',
            location: 'test 1'
        },
        {
            startTime: '5:55PM',
            location: 'test 2'
        }
    ]
};

describe('Interview List Component', () => {
    let mockFunction, wrapper;
    beforeEach(() => {
        mockFunction = jest.fn();

        const initialState = {
            interviews: {
                loading: false
            }
        };
        wrapper = setUp({ ...props, dispatch: mockFunction }, initialState);
    });

    describe('before finished loading', () => {
        it('does not render a list of interviews', () => {
            const initialState = {
                interviews: {
                    loading: true
                }
            };
            wrapper = setUp({ ...props, dispatch: mockFunction }, initialState);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when finished loading', () => {
        it('renders without error', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });
});
