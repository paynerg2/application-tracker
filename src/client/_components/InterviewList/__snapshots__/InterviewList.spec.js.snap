// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interview List Component before finished loading does not render a list of interviews 1`] = `
<Fragment>
  <InterviewContainer />
</Fragment>
`;

exports[`Interview List Component when finished loading renders without error 1`] = `
<Fragment>
  <InterviewContainer>
    <option>
      5:55PM, test 1
    </option>
    <option>
      5:55PM, test 2
    </option>
  </InterviewContainer>
</Fragment>
`;
